nilearn/decoding/tests/test_decoder.py

```diff
@@ -39,60 +50,140 @@
 from sklearn.preprocessing import StandardScaler
 from sklearn.svm import SVR, LinearSVC
 
-# Regression
-ridge = RidgeCV()
-svr = SVR(kernel="linear")
-# Classification
-svc = LinearSVC()
-logistic_l1 = LogisticRegression(penalty="l1")
-logistic_l2 = LogisticRegression(penalty="l2")
-ridge_classifier = RidgeClassifierCV()
-random_forest = RandomForestClassifier()
-
-dummy_classifier = DummyClassifier(random_state=0)
-dummy_regressor = DummyRegressor()
-
-regressors = {"ridge": (ridge, []), "svr": (svr, "C")}
-classifiers = {
-    "svc": (svc, "C"),
-    "logistic_l1": (logistic_l1, "C"),
-    "logistic_l2": (logistic_l2, "C"),
-    "ridge_classifier": (ridge_classifier, []),
-}
-# Create a test dataset
-rng = np.random.RandomState(0)
-X = rng.rand(100, 10)
-# Create different targets
-y_regression = rng.rand(100)
-y_classification = np.hstack([[-1] * 50, [1] * 50])
-
-
-def test_check_param_grid():
+N_SAMPLES = 100
+
+ESTIMATOR_REGRESSION = ["ridge", "svr"]
+
+
+def _make_binary_classification_test_data(n_samples=N_SAMPLES):
+    X, y = make_classification(
+        n_samples=n_samples,
+        n_features=125,
+        scale=3.0,
+        n_informative=5,
+        n_classes=2,
+        random_state=42,
+    )
+    X, mask = to_niimgs(X, [5, 5, 5])
+    return X, y, mask
+
+
+@pytest.fixture(scope="session")
+def rand_X_Y():
+    rng = np.random.RandomState(0)
+    X = rng.rand(N_SAMPLES, 10)
+    Y = np.hstack([[-1] * 50, [1] * 50])
+    return X, Y
+
+
+def _make_multiclass_classification_test_data(n_samples=200):
+    X, y = make_classification(
+        n_samples=n_samples,
+        n_features=125,
+        scale=3.0,
+        n_informative=5,
+        n_classes=4,
+        random_state=42,
+    )
+    X, mask = to_niimgs(X, [5, 5, 5])
+    return X, y, mask
+
+
+@pytest.fixture(scope="session")
+def dummy_binary_classification_data():
+    """Use for testting errors."""
+    return _make_binary_classification_test_data(n_samples=20)
```

Comment:
Why 'dummy' ?

Reply:
Mostly to distinguish from the fixture that is right below that is used for tests where a classification is actually performed.
This fixture on the other hand aims to return a very small data set because it will only be used for the tests that check the error handling like input validation.
Will try to add more comments but if you have a better name I am all ears.

Reply:
`tiny_binary_classification_data` sounds more accurate to me.

Reply:
accurate and cute: I like it.